junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java

```diff
@@ -821,6 +822,86 @@ public static void assertArrayEquals(Object[] expected, Object[] actual, Supplie
 		assertArrayEquals(expected, actual, new ArrayDeque<>(), messageSupplier);
 	}
 
+	// --- assertIterableEquals --------------------------------------------
+
+	/**
+	 * <em>Asserts</em> that {@code expected} and {@code actual} iterables are deeply equal.
+	 * <p>The check for deep equality works similarly to
+	 * {@link #assertArrayEquals(Object[], Object[])}, except rather than comparing elements by array
+	 * indexes, the following rule holds instead:
+	 * <ul>
+	 *   <li>If two iterables are encountered (including {@code expected} and {@code actual}), their
+	 *   iterators must return equal elements in the same order as each other. <strong>Note:</strong>
+	 *   this means that the iterables <em>do not</em> need to be of the same type.
+	 *   Example: <pre>   {@code
+	 *   import static java.util.Arrays.asList;
+	 *    . . .
+	 *   Iterable<Integer> i0 = asList(1, 2, 3, new ArrayList<>(asList(4, 5)));
```

Comment:
Yes, you make a good point! I'd be happy to make such a change as soon as I have the time.

Although, I do wonder if instead of "new LinkedList" if should be "new LinkedHashSet" or something non-List-y, so illustrate the point the iterables do _not_ have to be lists.